clippy_lints/src/attrs.rs

```diff
@@ -182,6 +182,29 @@ declare_clippy_lint! {
     "unknown_lints for scoped Clippy lints"
 }
 
+declare_clippy_lint! {
+    /// **What it does:** Checks for `warn`/`deny`/`forbid` attributes targeting the whole clippy::restriction category.
+    ///
+    /// **Why is this bad?** Restriction lints sometimes are in contrast with other lints or even go against idiomatic rust.
+    /// These lints should only be enabled on a lint-by-lint basis and with careful consideration.
+    ///
+    /// **Known problems:** None.
+    ///
+    /// **Example:**
+    /// Bad:
+    /// ```rust
+    /// #![deny(clippy::restriction)]
+    /// ```
+    ///
+    /// Good:
+    /// ```rust
+    /// #![deny(clippy::as_conversions)]
+    /// ```
+    pub BLANKET_CLIPPY_RESTRICTION_LINTS,
+    correctness,
```

Comment:
`correctness` is the wrong group, since enabling those lints doesn't have any influence on code semantics and doesn't change behavior.

I think `style` is the best group here, since "idomatic Clippy style" is to not enable the complete `restriction` group.

Reply:
Correctness lints are for "code that is just outright wrong or very very useless" so this fits IMO.

Reply:
I guess there are different ways to understand this, probably all of them valid. 

My take was that this is an incorrect usage of clippy, a footgun that can't be easily prevented due to `restriction` being a category like the others.

TBH I would be happy just with this being linted, as it's most of the time an error that is not hard to make (just requires not reading the README thoroughly).

Reply:
I'm okay with correctness or style.

Reply:
Yeah, I guess this comes down to everyone's interpretation of '_code that is outright wrong or very very useless_' from the `correctness` lint group description in the README.

Going from that description, it would certainly fit `correctness`. Still, I would tend to agree with @flip1995 here. IMO `correctness` is meant to warn about actual bugs in the code and nothing further. I don't think an incorrect Clippy usage is a bug in that sense.

Reply:
> IMO `correctness` is meant to warn about actual bugs in the code and nothing further. I don't think an incorrect Clippy usage is a bug in that sense.

I agree with this